esphome/components/grove_i2c_motor/grove_i2c_motor.cpp

```diff
@@ -0,0 +1,183 @@
+#include "grove_i2c_motor.h"
+#include "esphome/core/log.h"
+#include "esphome/core/hal.h"
+
+namespace esphome {
+namespace grove_i2c_motor {
+
+static const char *const TAG = "GroveMotorDriveTB6612FNG";
+
+static const uint8_t GROVE_MOTOR_DRIVER_I2C_CMD_BRAKE = 0x00;
+static const uint8_t GROVE_MOTOR_DRIVER_I2C_CMD_STOP = 0x01;
+static const uint8_t GROVE_MOTOR_DRIVER_I2C_CMD_CW = 0x02;
+static const uint8_t GROVE_MOTOR_DRIVER_I2C_CMD_CCW = 0x03;
+static const uint8_t GROVE_MOTOR_DRIVER_I2C_CMD_STANDBY = 0x04;
+static const uint8_t GROVE_MOTOR_DRIVER_I2C_CMD_NOT_STANDBY = 0x05;
+static const uint8_t GROVE_MOTOR_DRIVER_I2C_CMD_STEPPER_RUN = 0x06;
+static const uint8_t GROVE_MOTOR_DRIVER_I2C_CMD_STEPPER_STOP = 0x07;
+static const uint8_t GROVE_MOTOR_DRIVER_I2C_CMD_STEPPER_KEEP_RUN = 0x08;
+static const uint8_t GROVE_MOTOR_DRIVER_I2C_CMD_SET_ADDR = 0x11;
+
+void GroveMotorDriveTB6612FNG::dump_config() {
+  ESP_LOGCONFIG(TAG, "GroveMotorDriveTB6612FNG:");
+  LOG_I2C_DEVICE(this);
+}
+
+void GroveMotorDriveTB6612FNG::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up Grove Motor Drive TB6612FNG ...");
+  if (!this->standby()) {
+    this->mark_failed();
+    return;
+  }
+}
+
+bool GroveMotorDriveTB6612FNG::standby() {
+  uint8_t status = 0;
+  if (this->write_register(GROVE_MOTOR_DRIVER_I2C_CMD_STANDBY, &status, 1) != i2c::ERROR_OK) {
+    ESP_LOGW(TAG, "Set standby failed!");
+    this->status_set_warning();
+    return false;
+  }
+  return true;
+}
+
+bool GroveMotorDriveTB6612FNG::not_standby() {
+  uint8_t status = 0;
+  if (this->write_register(GROVE_MOTOR_DRIVER_I2C_CMD_NOT_STANDBY, &status, 1) != i2c::ERROR_OK) {
+    ESP_LOGW(TAG, "Set not standby failed!");
+    this->status_set_warning();
+    return false;
+  }
+  return true;
+}
+
+void GroveMotorDriveTB6612FNG::set_i2c_addr(uint8_t addr) {
+  if (addr == 0x00 || addr >= 0x80) {
+    return;
+  }
+  if (this->write_register(GROVE_MOTOR_DRIVER_I2C_CMD_SET_ADDR, &addr, 1) != i2c::ERROR_OK) {
+    ESP_LOGW(TAG, "Set new i2c address failed!");
+    this->status_set_warning();
+    return;
+  }
+  this->set_i2c_address(addr);
+}
+
+void GroveMotorDriveTB6612FNG::dc_motor_run(uint8_t chl, int16_t speed) {
```

Comment:
No need to shorten variable names, just makes it harder to understand
```suggestion
void GroveMotorDriveTB6612FNG::dc_motor_run(uint8_t channel, int16_t speed) {
```